src/package1/package1Version.ts

```diff
@@ -0,0 +1,146 @@
+  public static async display(connection: Connection, id: string): Promise<Package1Display[]> {
+    if (!id.startsWith('04t')) {
+      throw messages.createError('invalid04tId', [id]);
+    }
+    const query = `SELECT Id,MetadataPackageId,Name,ReleaseState,MajorVersion,MinorVersion,PatchVersion,BuildNumber FROM MetadataPackageVersion WHERE id = '${id}'`;
+    const results = (await connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query)).records;
+    return results.map((result) => ({
+      MetadataPackageVersionId: result.Id,
+      MetadataPackageId: result.MetadataPackageId,
+      Name: result.Name,
+      ReleaseState: result.ReleaseState,
+      Version: `${result.MajorVersion}.${result.MinorVersion}.${result.PatchVersion}`,
+      BuildNumber: result.BuildNumber,
+    }));
+  }
+
+  /**
+   * Lists package versions available in the org. If package ID is supplied, only list versions of that package,
+   * otherwise, list all package versions
+   *
+   * @param connection Connection to the org
+   * @param id: optional, if present, ID of package to list versions for (starts with 033)
+   * @returns Array of package version results
+   */
+  public static async list(connection: Connection, id?: string): Promise<Package1Display[]> {
+    if (id && !id?.startsWith('033')) {
+      // we have to check that it is present, and starts with 033
+      // otherwise, undefined doesn't start with 033 and will trigger this error, when it shouldn't
+      throw messages.createError('invalid033Id', [id]);
+    }
+    const query = `SELECT Id,MetadataPackageId,Name,ReleaseState,MajorVersion,MinorVersion,PatchVersion,BuildNumber FROM MetadataPackageVersion ${
+      id ? `WHERE MetadataPackageId = '${id}'` : ''
+    } ORDER BY MetadataPackageId, MajorVersion, MinorVersion, PatchVersion, BuildNumber`;
+
+    const queryResult = await connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query);
+    return queryResult.records?.map((record) => ({
+      MetadataPackageVersionId: record.Id,
+      MetadataPackageId: record.MetadataPackageId,
+      Name: record.Name,
+      ReleaseState: record.ReleaseState,
+      Version: `${record.MajorVersion}.${record.MinorVersion}.${record.PatchVersion}`,
+      BuildNumber: record.BuildNumber,
+    }));
+  }
+
+  /**
+   * Will create a PackageUploadRequest object based on the options provided, will poll for completion if pollingOptions are provided
+   *
+   * @param options: Package1VersionCreateRequest options for the new PackageUploadRequest to be created with
+   * @param pollingOptions: options to set frequency, and duration of polling. Default to not poll
+   */
+  public async create(
+    options: Package1VersionCreateRequest,
+    pollingOptions = { frequency: Duration.seconds(5), timeout: Duration.seconds(0) }
+  ): Promise<PackagingSObjects.PackageUploadRequest> {
+    const createRequest = await this.connection.tooling.sobject('PackageUploadRequest').create(options);
```

Comment:
The create is blindly using the options as crafted by the caller. I understand that the plugin command is vetting the inputs, but if used by an API consumer is the exception thrown by the tooling create sufficient to properly guide the caller to the mistake in input values?